projects/fw_102/src/main.c

```diff
@@ -0,0 +1,38 @@
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define COUNTER_DURATION 500
+
+typedef struct {
+  uint8_t counter_a;
+  uint8_t counter_b;
+  int turn;
+} Counters;
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
```

Comment:
Using the change I mentioned above, you can do this section in quite a few less lines